# Chapter 1 Exercises
In this notebook we will go through the exercises of chapter 1 of Introduction to Stochastic Processes with R by Robert Dobrow.

### 1.1
For the following scenarios identify the stochastic process {$X_t, t \in I$}, describing (i) $X_t$ in context (ii) state space, and (iii) index Set. State whether the state space and index are discrete or continuous.

    b) At the end of each year, a 4-year college student either advances in grade repeats the grade or drops out
    Answer:
let $X_t$ denote the state of the student at the end of the year $t$. This is a discrete space with $S=$ {Advances, Repeats, Drop out} for $I=${1,2,...}
    
    c) Seismologists record daily earthquake magnitudes in Chile. the largest recorded earthquake in history was the Valdivia, Chile earthquake on May 22 1960, which had a magnitude of 9.5 on Richter scale.
Answer:
ley $X_t$ the magnitude of earthquake t. and with continuous Space $S =$ (0,9.5] and $I=$ {0,1,2,...}
    
    d)Data are kept on circumferences of trees in an arboretum. The arboretum covers a two square-mile area.
    Answer:
ley $X_t$ the circumference of tree t. and with continuous Space $S =$ (0,max(circum)] and $I=$ {0,1,2,...}
    
    e) Starting Monday morning at 9 a.m., as students arrive to class, the teacher records the student arrival times. The class has 30 students and lasts for 60 minutes.
    Answer:
ley $X_t$ arrival time of student t. and with continuous Space $S =$ $[0,60]$ and $I=$ {0,1,2,...,30}
    
    f) A card player shuffles a standard deck of cards by the following method: the top card of the deck is placed somewhere in the deck at random. The player does this 100 times to mix up the deck.
    Answer:
ley $X_t$ the order of the cards at shuffle t. and with Discrete Space $S =$ 52! and $I=$ {0,1,2,...,100}
    

### 1.2
A regional Insurance company insures homeowners against flood damage. Half of their policyholders are in florida 30% in Louisiana, and 20% in Texas. Company Actuaries Inc. give the estimate table 1.1 for the probability that a policy holder will file a claim for flood damage over the next year. 

    a) find the probability that a random policyholder will file a claim for flood damage next year.
    b) a claim was filed. Find the probability that the policyholder is from texas 
|  | Table 1.1 Probability of claim for flood damage | |
| --- | :-: | --- |
| Florida | Louisiana | Texas |
| 0.03 | 0.015 | 0.02 |

#### Answer 
using the law of total probability, then we need to calculate 
$$P(claim)=\sum_{state} P(claim|State)*P(State)$$

In [2]:
prob_a = 0.5*0.03+0.3*0.015+0.2*0.02
prob_a

0.0235

In the second case we want to find the probability that it comes from texas given that a claim happened, then we want:
$$P(Texas|claim)=\frac{P(claim \cap Texas)}{P(claim)}=\frac{P(claim|Texas)P(Texas)}{P(claim)}$$

In [3]:
0.2*0.02/prob_a

0.1702127659574468

### 1.3
let $B_1,...,B_k$ be a partition of the sample space. For events A and C, prove the law of total probability for conditional probability.
$$P(A|C)=\sum_{i=1}^{k}P(A|B_i \cap C)P(B_i|C)$$

#### Answer

To answer this let's expand the right part of the equalityt
\begin{align}
\sum_{i=1}^{k}P(A|B_i \cap C)P(B_i|C) & = \sum_{i=1}^{k}\frac{P(A \cap B_i \cap C)}{P(B_i \cap C)} \frac{P(B_i \cap C)}{P(C)} \\
 & = \sum_{i=1}^{k}\frac{P(A \cap B_i \cap C)}{P(C)} \\
 & = \frac{1}{P(C)}\sum_{i=1}^{k}P(A \cap B_i \cap C) \\
 & = \frac{P(A \cap C)}{P(C)} = P(A|C)
\end{align}

### 1.4 
See exercise 1.2 Among all policyholders who live within five miles of the Atlantic Ocean, 75% live in florida, 20% in Louisiana and 5% live in Taxas. For thosewho live close to the ocean the probabilities of filing a claim increase as given in table 1.2.

Assume that the policyholder lives within five miles of the Atlantic Coast. Use the law of total probability fot conditional probability in Exercise 1.3 to find the chance that they will file a claim for flood damage next year.

|  | Table 1.2 Probability of claim for those within 5 miles | |
| --- | :-: | --- |
| Florida | Louisiana | Texas |
| 0.1 | 0.06 | 0.06 |

#### Answer
What we are looking for is:
$$P(claim|5m)=\sum_{state}P(claim | State \cap 5m)*P(State | 5m)$$

Note here that the partition is the state, that is why we are not looking to find the same result as before.

In [6]:
prob_claimG5m = 0.1*0.75+0.06*0.2+0.06*0.05
prob_claimG5m

0.09000000000000001

### 1.5
Two fais-sided dice are rolled let $X_1,X_2$ be the outcomes of the first and second die respectively.
    
- a) find the conditional distribution of $X_2$ given that $X_1+X_2=7$
- b) find the conditional distribution of $X_2$ given that $X_1+X_2=8$

#### Answer
we want $P(X_2|X_1+X_2=7)$ we can see that the values of $X_2$ are uniformly distributed {$1,2,3,4,5,6$} the reason for this is that if $X_1+X_2=7$, then if $X_1$ was 1, then $X_2$ needs to be 6, and so on with all the numbers that $X_1$ and $X_2$ can be.

in b) the solution comes from the same logic, but $X_2$ can never be 1 since $X_1$ would need to be 7 to make the sum 8, but 7 is not in the dice.

### 1.6
Bob has n coins in his pocket. One is two-headed, the rest are fair. A coin is picked at randomm flipped and shows heads. Find the probability that the coin is two headed.

#### Answer 
Let's say that H is the probability of obtaining a heads and S is the probability of getting a two sided head coin, then what the want is the following:
$$P(S|H)=\frac{P(S \cap H)}{P(H)}$$
but we can see that $P(S \cap H) = P(S)$ Since you are always going to get the heads if you have the two sided head coin, then we just need to look for the $P(S)$ and $P(H)$ and 
$$P(S) = 1/n$$
$$P(H) = \frac{1}{n}*1+\sum_{n-1}^{i=1}\frac{1}{n}*\frac{1}{2}=\frac{1}{n}+\frac{n-1}{2n} = \frac{n+1}{2n}$$
Then
$$P(S|H)=\frac{1/n}{n+1/(2n)}=\frac{2n}{n(n+1))}= \frac{2}{n+1}$$


In [32]:
def CoinFlipSim(n = 10, sims = 1000):
    # Note that for n = 10 the probability 2/11=0.1818
    import numpy as np
    results = 0
    heads = 0
    for i in range(sims):
        coin = np.random.randint(10)
        if coin == 0:
            results += 1
            heads += 1
        else:
            if np.random.random() < 0.5:
                heads += 1
    return results/heads
CoinFlipSim(sims=10000)

0.18511730205278593

### 1.7 
A rat is trapped in a maze with three doors and some hidden cheese. If the rat takes door one, he will wander around the maze for 2 minutes and return to where he started. If he takes door two, he will wander around the maze for 3 munutes and return to where he started. If he takes door three, he will find the cheese after one minute.  If the rat returns to where he started, then picks a new door. He picks a door by random, how long in average will it take it to get the cheese.

#### Answer
The key point here is to see that the expected value can be seen as a summ of the conditional expectations (due to the law of total expectation, then let $X$ be the time to reach the cheese and Y the door the rat chose we have:
\begin{align}
E(X) &= E(E(X|Y)) \\
 & = E(X|1)P(X=1)+E(X|2)P(X=2)+E(X|3)P(X=3) \\
 & = (2+E(X))1/3+(3+E(X))1/3+1/3 \\
 & = 2/3+1+1/3+2E(X)/3
\end{align}

Then $E(X)=6$, it is important to note that the 2 and 3 and 1 come from the time it takes the rat to go back or eat the cheese and since it goes back to the beginning, then counting from there we have again the $E(X)$.

this means that the rat will get to the cheese in 6 minutes 

In [89]:
def RatTimeMaze(sims = 10000):
    results = []
    for i in range(sims):
        time = ChooseDoor(0)
        results.append(time)
    return sum(results)/len(results)
        

def ChooseDoor(t):
    import numpy as np
    rand = np.random.random()
    if rand < 1/3:
        t=t+2
        return ChooseDoor(t)
    elif rand < 2/3:
        t=t+3
        return ChooseDoor(t)
    else:
        t=t+1
        return int(t)
RatTimeMaze()

6.0893

### 1.8 
A bag contains 1 red, 3 green and 5 yellow balls. A sample of four balls is picked. Let $G$ be the number of green balls in the sample. Let $Y$ be the number of yellow balls in the sample.

- a) Find the conditional probability mass function of $G$ given $Y=2$ assyming the sample is picked without replacement.
- b) Find the conditional probability mass function of $G$ given $Y=2$ assyming the sample is picked with replacement.

#### Answer
Since there have already been picked 2, then we only need to check for three cases $G=0,1,2$. In this case since we are looking for without replacement, then 
$$P(G=g|Y=2)=\frac{P(G=g \cap Y=2)}{P(Y=2)}$$
Let's calculate $P(Y=2)$
$$P(Y=2)==\frac{{5 \choose 2}{4 \choose 2}}{{9 \choose 4}}$$
and 
$$P(G=g \cap Y=2)==\frac{{5 \choose 2}{3 \choose g}{1 \choose 2-g}}{{9 \choose 4}}$$
when $1-g \geq 1$ 
Then we see that when $G=0$ this does not work, and both equations have the same denominator, then
$$P(G=1|Y=2)=\frac{{5 \choose 2}{3 \choose 1}{1 \choose 1}}{{5 \choose 2}{4 \choose 2}}$$
$$P(G=2|Y=2)=\frac{{5 \choose 2}{3 \choose 2}}{{5 \choose 2}{4 \choose 2}}$$

In [92]:
import math
def binom(n, k):
    return math.factorial(n) // math.factorial(k) // math.factorial(n - k)
p1 = binom(3,1)/binom(4,2) 
p2 = binom(3,2)/binom(4,2) 
p1, p2

(0.5, 0.5)

There fore the answer is 
$$P(G=1|Y=2)=0.5$$
$$P(G=2|Y=2)=0.5$$

For case b) we have that $P(Y=2)$ is a binomial distribution and $P(G=g \cap Y=2)$ is a multinomial distribution. Then
$$P(Y=2)={4 \choose 2}\biggl( \frac{5}{9} \biggr)^2 \biggl( \frac{4}{9} \biggr)^2$$
$$P(G=g \cap Y=2)=\frac{4!}{2!g!(2-g)!}\biggl( \frac{5}{9} \biggr)^2 \biggl( \frac{3}{9} \biggr)^g\biggl( \frac{1}{9} \biggr)^{2-g}$$
Then

In [101]:
import math 
py = binom(4,2)*(5/9)**2*(4/9)**2
p0 = (math.factorial(4)/(math.factorial(2)*math.factorial(2))*(5/9)**2*(1/9)**2)/py
p1 = (math.factorial(4)/(math.factorial(2))*(5/9)**2*(1/9)*(3/9))/py
p2 = (math.factorial(4)/(math.factorial(2)*math.factorial(2))*(5/9)**2*(3/9)**2)/py
p0,p1, p2

(0.0625, 0.375, 0.5625)

There fore the answer is 
$$P(G=0|Y=2)=0.0625$$
$$P(G=1|Y=2)=0.375$$
$$P(G=2|Y=2)=0.5625$$

### 1.9
Assume that X is uniformly distributed {$1,2,3,4$}. If X=x, then Y is uniformly distributed {$1,...,x$}. Find

- a)$P(Y=2|X=2)$
- b)$P(Y=2)$
- a)$P(X=2|Y=2)$
- a)$P(X=2)$
- a)$P(Y=2,X=2)$

#### Answer
a) $P(Y=2|X=2) = 1/2$ since Y is uniformly distributed {$1,2$}

b) using the law of total probability:
$$P(Y=2) = P(Y=2|X=2)P(X=2)+ P(Y=2|X=3)P(X=3)+P(Y=2|X=4)P(X=4) = 1/2*1/4+1/3*1/4+1/4+1/4$$
which makes
$$P(Y=2) = 0.270833  $$

c) $P(X=2|Y=2) = \frac{P(Y=2,X=2)}{P(Y=2)} = \frac{P(Y=2|X=2)P(X=2)}{P(Y=2)}$
Then:
$$P(X=2|Y=2) = \frac{1/8}{0.2708} = 0.46154414208174865$$

d) $$P(X=2) = 1/4$$

c) $$P(Y=2,X=2) = 1/8$$

In [2]:
1/8/0.27083

0.46154414208174865

### 1.10 
A die is rolled until a 3 occurs. By conditioning the outcome of the first roll, what is the probability that an even number of rolls is needed.


### Answer
Well, the condition just works for the case it is 3 the first roll in which case the probability is zero since it already happened. for the other cases the rolls should be independent of each other, so in this case the probability is the same as asking what is the probability of rolling a three in an odd number of rolls (since the first roll has already happened). 

in this case to happen on roll number two the probability is $1/6$, for it to happen in 4 rolls it is $5^2/6^3$, in 6 rolls it is $5^4/6^5$. Here the pattern then is that the probability of obtaining this is $\sum_{i=0}^\infty 5^{ni}/6^{2i+1}$ which is equal to:

$$p = \frac{1}{6*(1-5^2/6^2)} = \frac{6}{11}$$

If this question is framed to be answered as in question 1.9, then the answer is different (it is known right now that the answer we are looking for is 5/11). First the probability of getting evens can be seen as:

$$P(Evens) = P(Evens|R=3)*P(R=3)+P(Evens|R\neq 3) *P(R\neq 3)$$

It is important to note that since the events of the rolls are independent, then the probability of geting even given that the first roll was not even, then the probability is equal to the probability of getting the roll is odds (as discussed before). This means that if $P(Evens)=p$, then:

$$p = 0*1/6+(1-p) *5/6=> 11/6p=5/6 => p=5/11$$



### 1.11
Consider the gambler's ruin process where at each wager the gambler wins with probability p and loses with probability q=1-p. the gambler stops when reaching n or losing all their money. if they start with k, with 0<k<n. What is the probability of eventual ruin?. 

### Answer

Using the same logic as the one presented in the example we have that the $x_k=x_{k+1}p+x_{k-1}q$, but $x_{k+1}-x_k=(x_k-x_{k-1})p/q$. this means:

$$x_{k}-x_{k-1}=x_1(p/q)^k-1$$ 
and
$$x_{k}=\sum_{i=1}^k x_1(p/q)^k-1=x_1\frac{1-(p/q)^k}{1-(p/q)}$$

since we know that $x_n=1$, then:
$$1=x_1\frac{1-(p/q)^n}{1-(p/q)} => x_1=\frac{1-(p/q)}{1-(p/q)^n}$$

Which makes 

$$x_{k}=\frac{1-(p/q)^k}{1-(p/q)^n}$$


### 1.12
In n rolls of a fair die, let X be the number of times 1 is rolled, and Y the number of times 2 is rolled. Find the conditional distribution of X given Y=y.

### Answer
here $P(X=x)$ is a binomial distribution, in the case of making it conditional to Y, it still is a binomial distribution since the rolls are independent from each other, but what changes is the amount of times the experiment is done and the probability of getting y, since we know the amount of times 2 has appeared, this means there are only 5 possible outcomes on the die, this means that:
$$P(X=x|Y=y)={n-y \choose x} (1/5)^x (4/5)^{n-y-x}$$

### 1.13
Random Variables X and Y have probability density function :

$$f(x,y)=3y,   for 0<x<y<1$$

find the conditionals density functions

### Answer
Here we do the integrals and we get:

$f_Y(y)=\int_0^y 3y dx = 3y^2, for$ $0<y<1$

$f_X(x)=\int_x^1 3y dy = 3/2(1-x)^2, for$ $0<x<1$

then:
$$f_{X|Y}(x|y)=1/y,  0<x<y$$
$$f_{Y|X}(y|x)=2y/(1-x)^2, x<y<1$$

note that the first conditional density is a uniform density function.

### 1.14 
repear exercise 1.13, but with:

$$f(x,y)=4e^{-2x}, 0<y<x<\infty$$


### Answer
Doing the same as before

$f_X(x)=4xe^{-2x}, 0<x<\infty$

$f_y(y)=2e^{-2y}, 0<y<\infty$

then
$$f_{X|Y}(x|y)=2e^{-2x}/e^{-2y}, y<x<\infty$$
$$f_{Y|X}(y|x)=1/x, 0<y<x$$
The last density is a uniform density funciton.

### 1.15
Let X and Y be uniformly distributed in the circle of radius 1 at the origin. find the conditional density function of Y given X=x.



### Answer
The area of a cicle is defined around the bound x^2+y^2=1 and having a number picked x in the circle from 0-1, then it would delimit the y value to bounded by $-\sqrt{1-x^2}<y<\sqrt{1-x^2}$ and picking a value from here has to be uniform because of how the value is picked from the circle, then the result is given by:

$$f_{Y|X}(y|x)=\frac{1}{2\sqrt{1-x^2}}, -\sqrt{1-x^2}<y<\sqrt{1-x^2}$$

